tools/cldr-code/src/main/java/org/unicode/cldr/util/personname/PersonNameFormatter.java

```diff
@@ -292,10 +315,38 @@ public String format(NameObject nameObject) {
                 if (literal != null) {
                     result.append(literal);
                 } else {
-                    final String bestValue = nameObject.getBestValue(element.getModifiedField(), remainingModifers);
-                    if (!remainingModifers.isEmpty()) {
+                    String bestValue = nameObject.getBestValue(element.getModifiedField(), remainingModifers);
+                    if (bestValue == null) {
+                        // TODO Rich to handle collapsing whitespace / literals
+                        continue;
+                    } if (!remainingModifers.isEmpty()) {
                         // TODO Alex Apply unhandled modifiers algorithmically where possible
 
+                        // apply default algorithms
+                        // TODO specify order among modifiers
+
+                        for (Modifier modifier : remainingModifers) {
+                            switch(modifier) {
+                            case initial:
+                                BreakIterator breakIterator = characterBreakIterator;
+                                breakIterator.setText(bestValue);
+                                bestValue = bestValue.substring(0, breakIterator.next());
+                                // TODO Alex use initial pattern after defined
+                                break;
+                            case allCaps:
+                                bestValue = UCharacter.toUpperCase(formatterLocale, bestValue);
+                                break;
+                            case prefix:
+                                // TODO unhandled prefix is special; treat as if null
+                                // TODO if there is no plain, but there is a prefix and core, use that; otherwise use core
+                                break;
+                            case core:
+                            case informal:
+                                // no option, just fall back
```

Comment:
"Fall back" here means just return the unmodified version of the field (or at least skip this particular modifier), right?

Reply:
It means for most fields, skip the modifier. For the prefiix modifier, however, we actually want return an empty value, so that {surname-prefix} gets ignored if there is no explicit field.